model/src/test/kotlin/AbstractDependencyNavigatorTest.kt

```diff
@@ -0,0 +1,360 @@
+import io.kotest.matchers.collections.containExactly
+import io.kotest.matchers.collections.containExactlyInAnyOrder
+import io.kotest.matchers.collections.haveSize
+import io.kotest.matchers.collections.shouldContainExactlyInAnyOrder
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNot
+
+import java.io.File
+import java.time.Instant
+
+import org.ossreviewtoolkit.utils.test.readOrtResult
+
+/**
+ * A base class for tests of concrete [DependencyNavigator] implementations.
+ *
+ * The class is configured with an ORT result file that contains the expected results in a specific format. It then
+ * runs tests on the [DependencyNavigator] of this result and checks whether it returns the correct dependency
+ * information.
+ */
+abstract class AbstractDependencyNavigatorTest : WordSpec() {
+    /** The name of the result file to be used by all test cases. */
+    protected abstract val resultFileName: String
+
+    /**
+     * The name of the file with a result that contains issues. This is used by tests of the collectIssues() function.
+     */
+    protected abstract val resultWithIssuesFileName: String
+
+    private val testResult by lazy { readOrtResult(resultFileName) }
+
+    private val testProject by lazy { testResult.getProject(PROJECT_ID)!! }
+
+    protected val navigator by lazy { testResult.dependencyNavigator }
+
+    init {
+        "scopeNames" should {
+            "return the scope names of a project" {
+                navigator.scopeNames(testProject) should containExactlyInAnyOrder("compile", "test")
+            }
+        }
+
+        "directDependencies" should {
+            "return the direct dependencies of a project" {
+                navigator.directDependencies(testProject, "test")
+                    .map { it.id }.toList() should containExactly(
+                    Identifier("Maven:org.scalacheck:scalacheck_2.12:1.13.5"),
+                    Identifier("Maven:org.scalatest:scalatest_2.12:3.0.4")
+                )
+            }
+
+            "return an empty sequence for an unknown scope" {
+                navigator.directDependencies(testProject, "unknownScope").toList() should beEmpty()
+            }
+        }
+
+        "scopeDependencies" should {
+            "return a map with scopes and their dependencies for a project" {
+                val scopeDependencies = navigator.scopeDependencies(testProject)
+
+                scopeDependencies.keys should containExactlyInAnyOrder("compile", "test")
+
+                val compileDependencies = scopeDependencies["compile"].orEmpty()
+                compileDependencies should haveSize(17)
+                compileDependencies should containAll(
+                    Identifier("Maven:com.typesafe.akka:akka-actor_2.12:2.5.6"),
+                    Identifier("Maven:org.scala-lang:scala-reflect:2.12.2"),
+                    Identifier("Maven:org.scala-lang:scala-library:2.12.3")
+                )
```

Comment:
How about making the non-null check explicit via

                scopeDependencies["compile"].shouldNotBeNull {
                    this should haveSize(17)
                    this should containAll(
                        Identifier("Maven:com.typesafe.akka:akka-actor_2.12:2.5.6"),
                        Identifier("Maven:org.scala-lang:scala-reflect:2.12.2"),
                        Identifier("Maven:org.scala-lang:scala-library:2.12.3")
                    )
                }

Similar in more places below.